spec/rspec/support_spec.rb

```diff
@@ -180,5 +180,44 @@ def raise_instead_of_appending_to_failures_array
         expect { notify(error) }.to raise_instead_of_appending_to_failures_array
       end
     end
+
+    describe "warning notification" do
+      include RSpec::Support::Warnings
+
+      before { @warning_notifier = RSpec::Support.warning_notifier }
+      after  { RSpec::Support.warning_notifier = @warning_notifier }
+      let(:warnings) { [] }
+      let(:append_to_warnings_array_notifier) { lambda { |warning| warnings << warning } }
+
+      def append_to_array_instead_of_warning
+        change { warnings }.from([]).to([a_string_including('some warning')])
+      end
+
+      def warn_instead_of_appending_to_array
+        expect(::Kernel).to receive(:warn).with(a_string_including('some warning'))
+        avoid_changing { warnings }
+      end
+
+      it "defaults to warning with the provided text" do
+        expect(::Kernel).to receive(:warn).with(a_string_including('some warning'))
+        warning('some warning')
+      end
```

Comment:
I think I'd rather the `output().to_stderr` matcher was use rather than mocking `::Kernel.warn`.  Mocking seems unnecessary here.